discovery_engine/lib/src/domain/models/embedding.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:typed_data' show Float32List;
+
+import 'package:equatable/equatable.dart' show EquatableMixin;
+import 'package:hive/hive.dart' show BinaryReader, BinaryWriter, TypeAdapter;
+import 'package:xayn_discovery_engine/src/domain/repository/type_id.dart'
+    show embeddingAdapter;
+
+/// 1-Dimensional Embedding
+///
+/// Values are stored in native byte order in hive.
+class Embedding with EquatableMixin {
+  final Float32List values;
+
+  Embedding(this.values);
+
+  @override
+  List<Object?> get props => [values];
```

Comment:
```suggestion
  List<Object> get props => [values];
```

Reply:
The signature of `props` is `List<Object?>` not `List<Object>` you can change it to `List<Object>` as it's a sub-type, but why should we do so in this case?

Reply:
to help readability/analysis by having less type ambiguity/the most constrained type. i'd even argue that we should write `List<Float32List>`.

Reply:
Not in this case, the whole point of `probs` is to be as generic as possible i.e. `List<Object?>`,

Any place calling it will always treat it as `List<Object?>`.

There is no benefit gained and it will just obfuscate what this is doing.